### PR TITLE
add io.github.vegad.vnotes

### PR DESCRIPTION
**Pull request by vega-d**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

The app has dedicated access to `~/Nextcloud/Notes` to allow for desktop nextcloud notes integration.

### CODE REVIEW

**Comment by JakobDev** [comment]
You should use --filesystem=xdg-documents instead of --filesystem=~/Documents. The name of the Documents folder is translated, so this will not work on other languages than English.

You should also add the App ID to your PR title.

**Comment by vega-d** [comment]
> You should use --filesystem=xdg-documents instead of --filesystem=~/Documents. The name of the Documents folder is translated, so this will not work on other languages than English.

Thank you for noticing that, I hardcoded it for testing and forgot to change back



> You should also add the App ID to your PR title.

I didn't even realize it just took my commit message as PR name, oops

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: io.github.vegad.vnotes.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      - install -D src/settingsmenu.py /app/bin/settingsmenu.py
+      - install -D LICENSE.txt /app/bin/LICENSE.txt
+      - install -D io.github.vegad.vnotes.desktop /app/share/applications/io.github.vegad.vnotes.desktop
+      - install -D io.github.vegad.vnotes.appdata.xml /app/share/metainfo/io.github.vegad.vnotes.appdata.xml
+      - install -D icons/icon.svg /app/share/icons/hicolor/svg/apps/io.github.vegad.vnotes.svg
+      - install -D icons/icon_128.png /app/share/icons/hicolor/128x128/apps/io.github.vegad.vnotes.png
+      - install -D icons/icon_48.png /app/share/icons/hicolor/48x48/apps/io.github.vegad.vnotes.png
+    sources:
+      - type: git
+        url: https://github.com/vega-d/vnotes
```
[the remark this answers is not in the document]
Comment (vega-d) [reply io.github.vegad.vnotes.yml:26]: Do I need to include a specific commit there? My master branch gets new changes merged only when they are release-ready anyways

[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.github.vegad.vnotes.yml:26]: > you need a commit here.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.github.vegad.vnotes.yml:26]: the build must be reproducible, which mean that rebuilding from the same manifest should always lead to the same result.

**Review by hfiguiere — COMMENTED** [review]
Build will fail as the appdata file is lacking `releases` and `content_rating`

File: io.github.vegad.vnotes.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply io.github.vegad.vnotes.yml:26]: this is exactly what I said.

[the remark this answers is not in the document]
Comment (vega-d) [reply io.github.vegad.vnotes.yml:26]: I apologize for the confusion, I sent my previous message before yours loaded in.

**Comment by hfiguiere** [comment]
bot, build io.github.vegad.vnotes

**Comment by vega-d** [comment]
oh uh looks like I missed a tag in appstream, hold on

**Comment by vega-d** [comment]
wait where am I supposed to put that `<url>`

**Comment by vega-d** [comment]
Found it, `flatpak run --env=G_DEBUG=fatal-criticals org.freedesktop.appstream-glib validate builddir/*/share/appdata/io.github.vegad.vnotes.appdata.xml` passes with OK output

**Comment by barthalion** [comment]
bot, build io.github.vegad.vnotes

**Comment by vega-d** [comment]
the test build is working fine, except the desktop file is borked or something. If has a `--branch=master` argument but the app is in `test` branch so launching the test build from the gui doesn't work. Is that normal? If I drop the `--branch` argument from the desktop file it's working just fine

**Comment by barthalion** [comment]
It works fine for me. It's probably because you had a local build installed.

/merge

**Pull request closed without merging**